homie-device/src/lib.rs

```diff
@@ -489,29 +488,23 @@ impl DevicePublisher {
         value: impl Into<Vec<u8>>,
     ) -> Result<(), ClientError> {
         let topic = format!("{}/{}", self.device_base, subtopic);
-        let mut publish = Publish::new(topic, QoS::AtLeastOnce, value);
-        publish.retain = true;
-        self.requests_tx.send(publish.into()).await?;
-        Ok(())
+        self.client
+            .publish(topic, QoS::AtLeastOnce, true, value)
+            .await
     }
 
     async fn disconnect(&self) -> Result<(), ClientError> {
-        self.requests_tx.send(Request::Disconnect).await?;
-        Ok(())
+        self.client.disconnect().await
```

Comment:
Maybe just delete this method and use the one on client instead?

Reply:
Done.